codespell_lib/data/dictionary.txt

```diff
@@ -18740,12 +18810,13 @@ japanses->Japanese
 jaques->jacques
 javacript->javascript
 javascipt->javascript
-JavaSciript->JavaScript
+Javasciript->Javascript
```

Comment:
I remove the capitalization altogether. I don't think it makes sense to always capitalize programming languages. Most of the time in source code comments folks keep things pretty informal. My observation and also opinion

Reply:
In my opinion, those folks who make an effort for their code quality, documentation and spelling will use tools like codespell. Those folks who can't be bothered and want things to be informal, would mostly likely not be using quality checking tools.

The results from hunspell, aspell and ispell. All of them indicate a misspell and proposed `JavaScript`.
When I am typing this reply using Chrome browser, the browser recognises `Javascript` as misspelled, the word is underlined with a wavy red line. When I type in CamelCase, the wavy red line disappeared. 

IMO, the dictionary should hold true to what it should be. Not based on informal folks way of writing. People will then feel confident. @peternewman what do you think?

```text
### hunspell: /usr/share/hunspell: en_GB
& Javascript 3 0: JavaScript, Java script, Java-script
### hunspell: /usr/share/hunspell: en_US
& Javascript 3 0: JavaScript, Java script, Java-script
### aspell: /var/lib/aspell: en_GB
& Javascript 4 0: JavaScript, Java script, Java-script, JavaScript's
### aspell: /var/lib/aspell: en_US
& Javascript 4 0: JavaScript, Java script, Java-script, JavaScript's
### ispell: /usr/share/ispell: british-insane
& Javascript 3 0: JavaScript, Java script, Java-script
### ispell: /usr/share/ispell: american-insane
& Javascript 3 0: JavaScript, Java script, Java-script
```
```text
### look: marcoagpinto/aoo-mozilla-en-dict: en_GB (Marco Pinto)
JavaScript
### look: marcoagpinto/aoo-mozilla-en-dict: en_US (Kevin Atkinson)
JavaScript
```

Reply:
I understand your point.

Reply:
I've opened #2426 to discuss this in more detail, and unblock the merging of this PR because it's a separate issue altogether (preexisting entries are also called in to question).